grafanalib/zabbix.py

```diff
@@ -0,0 +1,706 @@
+        the returned value.
+    :param useCaptureGroups: defines if capture groups should be used during
+        metric query
+    """
+
+    application = attr.ib(default="", validator=instance_of(str))
+    expr = attr.ib(default="")
+    functions = attr.ib(default=attr.Factory(list))
+    group = attr.ib(default="", validator=instance_of(str))
+    host = attr.ib(default="", validator=instance_of(str))
+    intervalFactor = attr.ib(default=2, validator=instance_of(int))
+    item = attr.ib(default="", validator=instance_of(str))
+    itService = attr.ib(default="", validator=instance_of(str))
+    mode = attr.ib(default=ZABBIX_QMODE_METRICS, validator=instance_of(int))
+    options = attr.ib(default=attr.Factory(ZabbixTargetOptions),
+                      validator=instance_of(ZabbixTargetOptions))
+    refId = attr.ib(default="")
+    slaProperty = attr.ib(default=attr.Factory(dict))
+    textFilter = attr.ib(default="", validator=instance_of(str))
+    useCaptureGroups = attr.ib(default=False, validator=instance_of(bool))
+
+    def to_json_data(self):
+        obj = {
+            "application": ZabbixTargetField(self.application),
+            "expr": self.expr,
+            "functions": self.functions,
+            "group": ZabbixTargetField(self.group),
+            "host": ZabbixTargetField(self.host),
+            "intervalFactor": self.intervalFactor,
+            "item": ZabbixTargetField(self.item),
+            "mode": self.mode,
+            "options": self.options,
+            "refId": self.refId,
+        }
+        if self.mode == ZABBIX_QMODE_SERVICES:
+            obj["slaProperty"] = self.slaProperty,
+            obj["itservice"] = {"name": self.itService}
+        if self.mode == ZABBIX_QMODE_TEXT:
+            obj["textFilter"] = self.textFilter
+            obj["useCaptureGroups"] = self.useCaptureGroups
+        return obj
+
+
+@attr.s
+class ZabbixDeltaFunction(object):
+    """ZabbixDeltaFunction
+
+    Convert absolute values to delta, for example, bits to bits/sec
+    http://docs.grafana-zabbix.org/reference/functions/#delta
+    """
+    added = attr.ib(default=False, validator=instance_of(bool))
+
+    def to_json_data(self):
+        text = "delta()"
+        definition = {
+            "category": "Transform",
+            "name": "delta",
+            "defaultParams": [],
+            "params": []}
+        return {
+            "added": self.added,
+            "text": text,
+            "def": definition,
+            "params": [],
+        }
+
+
+@attr.s
+class ZabbixGroupByFunction(object):
+    """ZabbixGroupByFunction
+
+    Takes each timeseries and consolidate its points falled in given interval
+    into one point using function, which can be one of: avg, min, max, median.
+    http://docs.grafana-zabbix.org/reference/functions/#groupBy
+    """
+
+    __options = ("avg", "min", "max", "median")
+    __default_interval = "1m"
+    __default_function = "avg"
```

Comment:
Please don't use double underscores for privacy. Please just use a single underscore.